### PR TITLE
raises exceptions in normalization process when expected things are n…

### PR DESCRIPTION
**Pull request by mejackreed**
…ot found

Partially addresses #293. Though this may be controversial. I think documentation will also help. Additional enhancements could be made to the indexing code to print out the component in question?

Thoughts?

### CODE REVIEW

File: spec/lib/arclight/normalized_id_spec.rb [hunk 1]
```diff
@@ -24,8 +24,6 @@
   context 'when the id is nil' do
     let(:id) { nil }
 
-    it 'returns an empty string' do
-      expect(normalized_id).to eq ''
-    end
+    it { expect { normalized_id }.to raise_error Arclight::Exceptions::IDNotFound }
```
Comment (drh-stanford) [inline spec/lib/arclight/normalized_id_spec.rb]: the coverage report says that the `message` isn't being tested here. you can add a second parameter that tests the message.

File: spec/lib/arclight/normalized_title_spec.rb [hunk 2]
```diff
@@ -119,18 +119,6 @@
...
-
-  context 'no information' do
-    let(:title) { nil }
-    let(:date) { nil }
-    let(:id) { nil }
-
-    it 'does not know what to do' do
-      expect(normalized_title).to eq ''
-    end
+    it { expect { normalized_title }.to raise_error Arclight::Exceptions::TitleNotFound }
```
Comment (drh-stanford) [inline spec/lib/arclight/normalized_title_spec.rb]: the coverage report says that the `message` isn't being tested here. you can add a second parameter that tests the message.

**Comment by mejackreed** [comment]
@drh-stanford I've updated the PR addressing your comments.

**Comment by mejackreed** [comment]
I also indexed all of the demo eads using this and didn't raise any errors

**Comment by jkeck** [comment]
Hmm, so if I understand correctly, this will raise an error when there is no title instead of assigning the document id as the title?  Is that the expected behavior @anarchivist?

**Comment by mejackreed** [comment]
Yeah.. that is the new behavior. If the requirement is for each component to have unittitle and unitdate, I think that is approriate

**Comment by anarchivist** [comment]
👍  from me.

**Pull request merged**